### PR TITLE
Add ordering to menu items

### LINKED ISSUE #110: Ordering Items
Establish a way to order items. We can start with Core items and "Home".

In the slotFill landscape, should we establish a pattern for plugins to designate order?

### PR DESCRIPTION
**Pull request by joshuatf**
Fixes #110 

* Orders menu items
* Adds order to core menu items (WCA to come in follow-up in WCA repo)
* Adds args param to post type category item registration

### Screenshots
Screen Shot 2020-09-29 at 11 56 28 AM

### Testing
1. Check that items registered in this repo are in the correct order (see screenshot below).

Screen Shot 2020-09-29 at 12 01 46 PM

### CODE REVIEW

File: src/CoreMenu.php [hunk 1]
```diff
@@ -69,11 +69,11 @@ public function add_core_setting_items() {
...
-		Menu::add_post_type_category( 'shop_order' );
+		Menu::add_post_type_category( 'shop_order', array( 'order' => 20 ) );
 
 		// Products category.
 		Screen::register_post_type( 'product', null );
-		Menu::add_post_type_category( 'product' );
+		Menu::add_post_type_category( 'product', array( 'order' => 40 ) );
 
 		// Marketing category.
 		Menu::add_category(
```
Comment (psealock) [inline src/CoreMenu.php:79]: Marketing should get `'order' => 30` according to the screenshot.

**Review by psealock — APPROVED** [review]
Very nice, thanks @joshuatf. Just noting that Marketing ended up slightly off in my testing. Othewise, :shipit:

File: src/CoreMenu.php (hunk 1 above)
Comment (joshuatf) [reply src/CoreMenu.php:79]: Ah, I completely forgot this was in here.  Will update and also add a `@todo` to move this over to WCA where it should be registered.

Comment (joshuatf) [reply src/CoreMenu.php:79]: Actually, on second thought, I think this was registered here to add Coupons.  The marketing side in WCA should overwrite this with the new coupons menu.  Will make a follow-up to address this.

Added the menu order for the time being.

**Pull request merged**